src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -276,6 +282,9 @@ private void SetInputPullMode(int pinNumber, PinMode mode)
             for (int i = 0; i < 150; i++)
                 ;
 
+            // Spec calls to reset clock after the control signal
+            // Since context switch between those two instructions can potentially
+            // change pull up/down value we reset the clock first.
```

Comment:
Talking about context switches... What happens if two threads try to simultaneously change two (distinct) pins at the same time? Do we need a global lock around this method?